# How many total gifts do you get in the 12 Days of Christmas song?

For each day of the $\text{"Twelves Days of Christmas"}$ we receive __a new gift__, and _for each gift_,   
we receive a __quantity relative to a strictly increasing count of which of the 12 days it is__. 
   
However, it is important to observe that, _for each concurrent day_, the __new gifts _compound_ with all the gifts that came before it__, as follows:
   
|Day of Christmas| # of Gifts Received |
|:---:|:---:|
|First| 1 |
|Second| 2 + _(# Gifts Received First Day)_ |
|Third | 3 + _(# Gifts Received Second Day)_ + _(# Gifts Received First Day)_ |
| ...  | ... |
|Twelfth| 12 + _(# Gifts Received Eleventh Day)_ + _(# Gifts Received Tenth Day)_ + $\dots$ + _(# Gifts Received First Day)_  |   

<br>

A more mathematical representation which expresses the accumulation of gifts received is:
$[1] + [2 + 1] + [3 + 2 + 1]\,+\,\dots\,+\,[12 + 11 + ... + 1]$

<br>

We can represent something similar to this using ```python```.  
As a first step, we can express the accumulation as a _multi-dimensional list_ whose rows each represent a _consecutively increasing range of days_:

In [1]:
[ list( range(1, day + 1) ) for day in range(1, 13) ]

[[1],
 [1, 2],
 [1, 2, 3],
 [1, 2, 3, 4],
 [1, 2, 3, 4, 5],
 [1, 2, 3, 4, 5, 6],
 [1, 2, 3, 4, 5, 6, 7],
 [1, 2, 3, 4, 5, 6, 7, 8],
 [1, 2, 3, 4, 5, 6, 7, 8, 9],
 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11],
 [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]]

<br>

Going further, instead of merely listing them, we could rather __sum up all the days respective to each row__ with only a small modification: 

In [2]:
[ sum( range(1, day + 1) ) for day in range(1, 13) ]

[1, 3, 6, 10, 15, 21, 28, 36, 45, 55, 66, 78]

<br>

Finally, since we have, in essence, now _aggregated each row in the multi-dimensional list_ into a _single, calculated value_,   
We may now __take the sum of this aggregated result__ to expose the solution, as follows: 
 

In [3]:
sum( sum( range(1, day + 1) ) for day in range(1, 13) )

364

<br>

$Conclusively$,  
  
The ```python``` code executed above thereby validates the mathematical assertion that:  

$$(1) + (2 + 1) + (3 + 2 + 1) + \dots + (12 + 11 + \dots + 1)$$
$$=$$
$$1 + 3 + 6 + 10 + 15 + 21 + 28 + 36 + 45 + 55 + 66 + 78$$
$$=$$
$$\Large{364}$$